src/ActionList.fs

```diff
@@ -0,0 +1,54 @@
+module Polaris.ActionList
+
+open Fable.React
+open Fable.Core
+open Fable.Core.JsInterop
+open Polaris.Shared
+
+type [<StringEnum>] [<RequireQualifiedAccess>] ItemDescriptorBadgeStatus =
+    |  [<CompiledName "new">] New
+
+type [<RequireQualifiedAccess>] ActionListItemDescriptorBadge(content, ?status) =
+    member __.Content : string = content
+    member __.Status : ItemDescriptorBadgeStatus option = status
+
+type [<RequireQualifiedAccess>] ActionListItemDescriptor(content, ?accessibilityLabel, ?active, ?badge, ?destructive, ?disabled, ?ellipsis, ?external, ?helpText, ?icon, ?id, ?image, ?role, ?target, ?url, ?onAction) =
+    member __.AccessibilityLabel : string option = accessibilityLabel
+    member __.Active : bool option = active
+    member __.Badge : ActionListItemDescriptorBadge option = badge
+    member __.Content : string = content
+    member __.Destructive : bool option = destructive
+    member __.Disabled : bool option = disabled
+    member __.Ellipsis : bool option = ellipsis
+    member __.External : bool option = external
+    member __.HelpText : string option = helpText
+    member __.Icon : BundledIcon option = icon
+    member __.Id : string option = id
+    member __.Image : string option = image
+    member __.Role : string option = role
+    member __.Target : LinkTarget option = target
+    member __.Url : string option = url
+    member __.OnAction : (unit -> unit) option = onAction
+
+type [<RequireQualifiedAccess>] ActionListSection(items, title) =
+    member __.Items : ActionListItemDescriptor array = items
+    member __.Title : string = title
+
+type RequiredActionListProps = {
+    Items : ActionListItemDescriptor array
```

Comment:
What I've seen other people using instead of `array` is `ResizeArray<ActionListItemDescriptor>` I'm not sure why

Reply:
IMO this is most likely not needed since we consume this with Elmish, so the array should never change except when its completely rebuilt.

Reply:
isn't this object being passed to Polaris in the end? I'm asking because I'm not sure if Fable differentiates between them when transpiling.

Reply:
It is passed to the polaris component down the line,

I played with the REPL a bit to understand the difference, given:
```
type a = {
    arr: string array
    arr2: ResizeArray<string>
}

let b = {
    arr = [|"test"|]
    arr2 = ResizeArray([|"test2"|])
}
```

We get the following JS code:
```
export const b = new a(["test"], Array.from(["test2"]));
```

I have tested both in the JS console, there seem to be no real difference between to two, so my take would be that with elmish specifically, this will never be an issue because of state changes, but if you were to do a simple fable app and manage state update by yourself, using the resize array could prevent you from running into some weird issues where it updated but not created a complete new copy (new memory ref)